packages/back-end/src/queryRunners/QueryRunner.ts

```diff
@@ -491,20 +493,22 @@ export abstract class QueryRunner<
     process: (rows: Rows) => ProcessedRows,
     currentTimeout: number = INITIAL_CONCURRENCY_TIMEOUT
   ): Promise<void> {
-    // If too many queries are running against the datastore, use capped exponential backoff to wait until they've finished
+    // If too many queries are running against the datastore
+    // use capped exponential backoff to wait until they've finished
     const concurrencyLimitReached = await this.concurrencyLimitReached();
     if (concurrencyLimitReached) {
       logger.debug(
         `${doc.id}: Query concurrency limit reached, waiting ${currentTimeout} before retrying`
       );
-      this.runCallbacks[doc.id] = { run, process };
```

Comment:
This doesn't need to be repeated each time the query is attempted so it's been moved to before the query is queued